app/utils/smartapi/data_processor.py

```diff
@@ -35,3 +35,40 @@ def process_smart_api_stock_data(
         symbol_token=stock_price_data["symboltoken"],
         prev_day_close=stock_price_data["close"],
     )
+
+
+def process_smart_api_historical_stock_data(
+    historical_stock_data: tuple,
+) -> List[HistoricalStockPriceInfo]:
+    """
+    Processes the data from the SmartAPI and returns the processed data.
+
+    Parameters:
+    -----------
+    stock_symbol: `str`
+        stock symbol.
+    historical_stock_data: `List[List[Any]]`
+        The data from the SmartAPI to be processed.
+
+    Return:
+    -------
+    List[HistoricalStockPriceInfo]
+        The processed data from the SmartAPI as a list of HistoricalStockPriceInfo.
+    """
+    processed_historical_stock_data = []
+    for stock_data in historical_stock_data[0]:
+        if len(stock_data) == 0:
+            continue
+        processed_historical_stock_data.append(
+            HistoricalStockPriceInfo(
+                timestamp=stock_data[0],
+                open=stock_data[1],
+                close=stock_data[2],
+                low=stock_data[3],
+                high=stock_data[4],
+                volume=stock_data[5],
+                stock_symbol=historical_stock_data[1],
+                candle_interval=historical_stock_data[2],
+            )
+        )
+    return processed_historical_stock_data
```

Comment:
Use list comprehension here which is faster.
![image](https://github.com/Munikumar09/market_data_api/assets/116535095/ff7e00e2-ec0f-4ab7-a923-0056efe4144e)